deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -0,0 +1,120 @@
+from deepchem.feat.base_classes import MolecularFeaturizer
+from deepchem.utils.molecule_feature_utils import one_hot_encode
+import numpy as np
+from rdkit import Chem
+from sklearn.metrics import pairwise_distances
+
+
+class MATFeaturizer(MolecularFeaturizer):
+  """
+  This class is a featurizer for the Molecule Attention Transformer [1]_.
+  The featurizer accepts an RDKit Molecule, and 2 booleans (add_dummy_node and one_hot_formal_charge) as arguments.
+  The returned value is a tuple which consists of molecular graph descriptions:
+    - Node Features
+    - Adjacency Matrix
+    - Distance Matrix
+  """
+
+  def __init__(
+      self,
+      mol: Chem.rdchem.Mol,
```

Comment:
This should be type `RDKITMol`. We can't directly use rdkit types here since it's an optional dependency. Check out the type signatures of the other featurizers for examples

Reply:
Done!